cfme/tests/cloud_infra_common/test_cloud_init_provisioning.py

```diff
@@ -98,3 +110,81 @@ def test_provision_cloud_init(appliance, request, setup_provider, provider, prov
     with ssh.SSHClient(hostname=connect_ip, username=provisioning['ci-username'],
                        password=provisioning['ci-pass']) as ssh_client:
         wait_for(ssh_client.uptime, num_sec=200, handle_exception=True)
+
+
+@pytest.mark.rhv3
+@pytest.mark.uncollectif(lambda provider, appliance:
+    not provider.one_of(RHEVMProvider) or appliance.version < '5.10.0.12',
+    reason='Required customization template is only available in CFME 5.10.0.12 or greater.')
+def test_provision_cloud_init_payload(appliance, request, setup_provider, provider, provisioning,
+                                      vm_name):
+    """
+    Tests that options specified in VM provisioning dialog in UI are properly passed as a cloud-init
+    payload to the newly provisioned VM.
+
+    Metadata:
+        test_flag: cloud_init, provision
+    """
+    image = provisioning.get('ci-image', None)
+    if not image:
+        pytest.skip('No ci-image found in provider specification.')
+    note = ('Testing provisioning from image {} to vm {} on provider {}'.format(
+        image, vm_name, provider.key))
+    logger.info(note)
+
+    ci_payload = {
+        'root_password': 'mysecret',
+        'address_mode': 'Static',
+        'hostname': 'cimachine',
+        'ip_address': '169.254.0.1',
+        'subnet_mask': '29',
+        'gateway': '169.254.0.2',
+        'dns_servers': '169.254.0.3',
+        'dns_suffixes': 'virt.lab.example.com',
+        'custom_template': {'name': 'oVirt cloud-init'}
+    }
+
+    inst_args = {
+        'request': {'notes': note},
+        'customize': {'customize_type': 'Specification'},
+        'template_name': image
+    }
+
+    inst_args['customize'].update(ci_payload)
+    logger.info('Instance args: {}'.format(inst_args))
+
+    # Provision VM
+    collection = appliance.provider_based_collection(provider)
+    instance = collection.create(vm_name, provider, form_values=inst_args)
+    request.addfinalizer(instance.cleanup_on_provider)
+    provision_request = provider.appliance.collections.requests.instantiate(vm_name,
+                                                                            partial_check=True)
+    provision_request.wait_for_request()
+
+    connect_ip = wait_for(find_global_ipv6, func_args=[instance], num_sec=600, delay=20).out
+    logger.info('Connect IP: {}'.format(connect_ip))
+
+    # Connect to the newly provisioned VM
+    with ssh.SSHClient(hostname=connect_ip,
+                       username='root',
+                       password=ci_payload['root_password']) as ssh_client:
+
+        # Check that correct hostname has been set
+        _, stdout, stderr = ssh_client.exec_command('hostname')
+        error, output = stderr.read(), stdout.read()
+        assert error == '', 'There was an error during getting hostname: {}'.format(error)
```

Comment:
Question:
Instate of using `exec_command` why are you not going with `run_command`?. It will help you to assert `success` and you will get direct `output`. Check small example..

```In [25]: x = ssh.SSHClient(hostname=appliance.hostname, username='root', password='smartvm')

In [26]: x.connect()

In [27]: x.connected
Out[27]: True

In [28]: host_name = x.run_command('hostname')

In [29]: host_name.success
Out[29]: True

In [30]: host_name.output
Out[30]: u'dhcp-8-196-130.cfme2.lab.eng.rdu2.redhat.com\n' ```

Reply:
Nice hint, thank you!